### PR TITLE
Unittest

### PR DESCRIPTION
**Pull request by amyxchen**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -75,8 +75,7 @@ def finalize_options(self):
     self.test_args = []
     self.test_suite = True
 
-  def run(self):
-    test.run(self)
```
Comment (jettisonjoe) [inline setup.py]: Is test.run not needed?

**Comment by fahhem** [comment]
Ah, this is a great change! It overwrites only the running of tests, not the rest of it.

File: setup.py (hunk 1 above)
Comment (fahhem) [inline setup.py]: test.run eventually runs this run_tests function

Comment (jettisonjoe) [inline setup.py]: Got it. Just found that in setuptools' source.

**Comment by jettisonjoe** [comment]
Merging.

**Pull request merged**